alacritty_terminal/src/term/mod.rs

```diff
@@ -1743,25 +1745,42 @@ impl<T: EventListener> Handler for Term<T> {
         trace!("Clearing line: {:?}", mode);
 
         let cursor = self.grid.cursor;
-        match mode {
+        let num_cols = self.grid.num_cols();
+        let selection_range = self.selection.as_ref().and_then(|s| s.to_range(&self));
+        let buffer_cursor = self.grid.buffer_cursor();
+
+        let cols = match mode {
             ansi::LineClearMode::Right => {
                 let row = &mut self.grid[cursor.point.line];
-                for cell in &mut row[cursor.point.col..] {
+                let cols = cursor.point.col..num_cols;
+                for cell in &mut row[cols.clone()] {
```

Comment:
From here on out all branches basically do the same thing. So if we need to compute the cols anyways and return them, we can probably extract everything other than just the cols computation to after the match?

If `contains_cols_on_buffer_line` takes a reference to a range, we should also be able to get rid of the `clone`s? Even though the clones are extremely cheap of course.